src/ports/utils.ts

```diff
@@ -57,3 +66,69 @@ export async function processRequestError(action: string, response: Response) {
     `Error ${action}, the server responded with: ${response.status}`
   )
 }
+
+export function isPaginated<T>(
+  data: T | PaginatedResponse<T>
+): data is PaginatedResponse<T> {
+  return (data as PaginatedResponse<T>).results !== undefined
+}
+
+export async function queryMultipleTimesWhenExceedingUrlLimit<T>(
+  baseUrl: string,
+  queryParameterName: string,
+  queryParameterValues: string[],
+  fetchComponent: IFetchComponent
+): Promise<T[]> {
+  const limit = pLimit(MAX_CONCURRENT_REQUEST)
+
+  // Build URLs to get all the queried Picks Stats
+  let urls: string[] = []
+  let url = baseUrl
+
+  const alreadyHasQueryParams = baseUrl.includes('?')
+
+  for (let value of queryParameterValues) {
+    const param =
+      (alreadyHasQueryParams ? `&` : '') + `${queryParameterName}=${value}`
```

Comment:
This should, if it's the first element of the array, choose between using `?` or `&` depending on if it already has query parameters or not, and then, for all of the other cases, append `&`.
As it is here, if the base url doesn't have query parameters, it will concatenate all of the query parameters without the use of &.

Reply:
Done!